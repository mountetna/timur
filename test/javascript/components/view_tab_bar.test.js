import React from 'react';
import {render, screen, waitFor, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ViewTabBar from '../../../lib/client/jsx/components/browser/view_tab_bar';

describe('ViewTabBar', () => {
  it('renders without a regex', async () => {
    const {asFragment} = render(
      <ViewTabBar
        view={{
          tabs: [
            {
              name: 'overview',
              title: 'Overview',
              description: null,
              panes: [
                {
                  name: 'pane1',
                  items: [
                    {
                      name: 'attribute1',
                      type: 'markdown',
                      text: 'This is a test'
                    }
                  ],
                  title: 'Pane1'
                }
              ]
            },
            {
              name: 'sequencing',
              title: 'Sequencing',
              description: null,
              panes: [
                {
                  name: 'pane1',
                  items: [
                    {
                      name: 'attribute1',
                      type: 'markdown',
                      text: 'This is a test 2'
                    }
                  ],
                  title: 'Pane1'
                }
              ]
            }
          ]
        }}
        recordName='PROJ1-HS1'
      />
    );

    await waitFor(() => screen.getByText('Overview'));

    expect(screen.queryByText(/Overview/)).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with a regex that matches the record_name', async () => {
    const {asFragment} = render(
      <ViewTabBar
        view={{
          tabs: [
            {
              name: 'overview',
              title: 'Overview',
              description: null,
              panes: [
                {
                  name: 'pane1',
                  items: [
                    {
                      name: 'attribute1',
                      type: 'markdown',
                      text: 'This is a test'
                    }
                  ],
                  title: 'Pane1'
                }
              ],
              regex: 'PROJ1'
            },
            {
              name: 'sequencing',
              title: 'Sequencing',
              description: null,
              panes: [
                {
                  name: 'pane1',
                  items: [
                    {
                      name: 'attribute1',
                      type: 'markdown',
                      text: 'This is a test 2'
                    }
                  ],
                  title: 'Pane1'
                }
              ]
            }
          ]
        }}
        recordName='PROJ1-HS1'
      />
    );

    await waitFor(() => screen.getByText('Overview'));

    expect(screen.queryByText(/Overview/)).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  it('does not render if regex false', () => {
    const {container} = render(
      <ViewTabBar
        view={{
          tabs: [
            {
              name: 'overview',
              title: 'Overview',
              description: null,
              panes: [
                {
                  name: 'pane1',
                  items: [
                    {
                      name: 'attribute1',
                      type: 'markdown',
                      text: 'This is a test'
                    }
                  ],
                  title: 'Pane1'
                }
              ],
              regex: 'SCG'
            },
            {
              name: 'sequencing',
              title: 'Sequencing',
              description: null,
              panes: [
                {
                  name: 'pane1',
                  items: [
                    {
                      name: 'attribute1',
                      type: 'markdown',
                      text: 'This is a test 2'
                    }
                  ],
                  title: 'Pane1'
                }
              ]
            }
          ]
        }}
        recordName='PROJ1-HS1'
      />
    );
    expect(container.innerHTML).toEqual('<div style="display: none;"></div>');
  });
});
