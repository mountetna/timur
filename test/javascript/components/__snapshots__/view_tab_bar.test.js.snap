// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewTabBar renders with a regex that matches the record_name 1`] = `
<DocumentFragment>
  <div
    class="tabbar"
  >
    <div
      class="spacer1"
    />
    <div
      class="tab"
    >
      Overview
    </div>
    <div
      class="tab"
    >
      Sequencing
    </div>
    <div
      class="spacer2"
    />
  </div>
</DocumentFragment>
`;

exports[`ViewTabBar renders without a regex 1`] = `
<DocumentFragment>
  <div
    class="tabbar"
  >
    <div
      class="spacer1"
    />
    <div
      class="tab"
    >
      Overview
    </div>
    <div
      class="tab"
    >
      Sequencing
    </div>
    <div
      class="spacer2"
    />
  </div>
</DocumentFragment>
`;
